x/stake/endblock.go

```diff
@@ -10,9 +10,15 @@ import (
 )
 
 func EndBlocker(ctx sdk.Context, k keeper.Keeper) (validatorUpdates []abci.ValidatorUpdate, completedUbds []types.UnbondingDelegation) {
+	// only change validator set in breath block after BEPHHH
 	var events sdk.Events
-	_, validatorUpdates, completedUbds, _, events = handleValidatorAndDelegations(ctx, k)
 	var csEvents sdk.Events
+	if !sdk.IsUpgrade(sdk.BEPHHH) {
```

Comment:
if `!sdk.IsUpgrade(sdk.BEPHHH)`, the `events` here seem to be emitted twice.

Reply:
Yes. It's a duplication introduced by merging the native staking code. Fixed.